### PR TITLE
Gen 4: Fix Non-Multitype Arceus Gain/Loss of Plates Interaction

### PR DESCRIPTION
**Pull request by Caznovia**
https://www.smogon.com/forums/threads/gen-4-arceus-without-multitype-should-be-able-to-gain-lose-plates.3756056/

### CODE REVIEW

File: data/mods/gen4/items.ts [hunk 1]
```diff
@@ -113,15 +113,45 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	dracoplate: {
+		inherit: true,
+		onTakeItem(item, pokemon, source) {
+			return true;
+		}
```
[the remark this answers is not in the document]
Comment (Karthik99999) [reply data/mods/gen4/items.ts:120]: Actually scratch that, the function needs to be changed to check if either pokemon has multitype.

[the remark this answers is not in the document]
Comment (Caznovia) [reply data/mods/gen4/items.ts]: The function shouldn't need to be changed to check for Multitype - testing the code on my server shows that Trick/Fling still fail if a Multitype Pokemon gains/loses a plate.

[the remark this answers is not in the document]
Comment (Karthik99999) [reply data/mods/gen4/items.ts]: mb, just saw that the failure for that is inside trick/switcheroo.

File: data/mods/gen4/items.ts [hunk 2]
```diff
@@ -306,6 +342,10 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			},
 		},
 	},
+	mindplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:347]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 3]
```diff
@@ -250,6 +282,10 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	meadowplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:287]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 4]
```diff
@@ -156,10 +176,22 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
...
+		inherit: true,
+		onTakeItem: true, 
+	},
 	ironball: {
 		inherit: true,
 		onEffectiveness() {},
 	},
+	ironplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:193]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 5]
```diff
@@ -156,10 +176,22 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 		inherit: true,
 		isNonstandard: null,
 	},
+	icicleplate: {
+		inherit: true,
+		onTakeItem: true, 
+	},
+	insectplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:185]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 6]
```diff
@@ -156,10 +176,22 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 		inherit: true,
 		isNonstandard: null,
 	},
+	icicleplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:181]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 7]
```diff
@@ -113,15 +113,35 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
...
+		onTakeItem: true, 
+	},
 	flameorb: {
 		inherit: true,
 		onResidualOrder: 10,
 		onResidualSubOrder: 20,
 	},
+	flameplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:143]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 8]
```diff
@@ -113,15 +113,35 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
...
+		inherit: true,
+		onTakeItem: true, 
+	},
 	fastball: {
 		inherit: true,
 		isNonstandard: null,
 	},
+	fistplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:134]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 9]
```diff
@@ -113,15 +113,35 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
...
+		inherit: true,
+		onTakeItem: true, 
+	},
+	dreadplate: {
+		inherit: true,
+		onTakeItem: true, 
+	},
+	earthplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:126]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 10]
```diff
@@ -113,15 +113,35 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	dracoplate: {
+		inherit: true,
+		onTakeItem: true, 
+	},
+	dreadplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:122]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 11]
```diff
@@ -113,15 +113,35 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	dracoplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:118]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 12]
```diff
@@ -325,6 +365,18 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	skyplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:370]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 13]
```diff
@@ -325,6 +365,18 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	skyplate: {
+		inherit: true,
+		onTakeItem: true, 
+	},
+	splashplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:374]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 14]
```diff
@@ -325,6 +365,18 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
...
+		inherit: true,
+		onTakeItem: true, 
+	},
+	splashplate: {
+		inherit: true,
+		onTakeItem: true, 
+	},
+	spookyplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:378]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 15]
```diff
@@ -342,6 +394,10 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 		onResidualOrder: 10,
 		onResidualSubOrder: 20,
 	},
+	stoneplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:399]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 16]
```diff
@@ -355,6 +411,10 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 		onResidualOrder: 10,
 		onResidualSubOrder: 20,
 	},
+	toxicplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:416]:
```suggestion
		onTakeItem: true,
```

File: data/mods/gen4/items.ts [hunk 17]
```diff
@@ -364,6 +424,10 @@ export const Items: import('../../../sim/dex-items').ModdedItemDataTable = {
 			}
 		},
 	},
+	zapplate: {
+		inherit: true,
+		onTakeItem: true,
```
Comment (KrisXV) [inline data/mods/gen4/items.ts:429]:
```suggestion
		onTakeItem: true,
```

**Pull request merged**